### PR TITLE
Resolve absolute paths based on directory option

### PR DESCRIPTION
**Pull request by pahen**
Part of fixing https://github.com/mrjoelkemp/node-dependency-tree/issues/40

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -30,20 +32,34 @@ module.exports = function(options) {
   debug('filename: ', filename);
 
   if (typeof config === 'string') {
-    configPath = config;
+    configPath = path.dirname(config);
     config = module.exports._readConfig(config);
     debug('converting given config file ' + configPath + ' to an object:\n', config);
   }
 
+  if (configPath && !fs.statSync(configPath).isDirectory()) {
```
Comment (pahen) [inline index.js:40]: This is needed for the existing tests to pass since configPath is set to a file and not a directory.

File: index.js [hunk 2]
```diff
@@ -20,7 +22,7 @@ var requirejs = require('requirejs');
 module.exports = function(options) {
   // If you want to supply a preparsed config object without a configPath, then
   // we have no way of knowing what the baseUrl should be.
-  var configPath = options.configPath || options.filename;
+  var configPath = options.configPath;
```
Comment (pahen) [inline index.js:25]: I did some changes in how configPath is set so it's easier to follow what's happening.

File: index.js [hunk 3]
```diff
@@ -30,20 +32,34 @@ module.exports = function(options) {
...
   }
 
-  var resolutionDirectory = path.dirname(configPath);
-  debug('module resolution directory: ' + resolutionDirectory);
+  var resolutionDirectory;
+
+  if (configPath) {
+    resolutionDirectory = configPath;
+    debug('module resolution directory (based on configPath): ' + resolutionDirectory);
+  } else if (options.directory && depPath.charAt(0) !== '.') {
```
Comment (pahen) [inline index.js:56]: This should maybe be depPath[0] instead ..

**Comment by mrjoelkemp** [comment]
Awesome work man! Thanks for contributing. The changes you made are great.

I made some tiny changes (updated Readme, added an npmignore while I was there).

This is now in the 4.0.3 patch release. 

🎉

**Comment by pahen** [comment]
Sweet! Glad you liked it :) Can you please publish the latest release to NPM too?

**Comment by mrjoelkemp** [comment]
Whoops! Done :)

**Pull request closed without merging**